src/sidebar/components/MentionSuggestionsPopover.tsx

```diff
@@ -0,0 +1,71 @@
+import { Popover } from '@hypothesis/frontend-shared';
+import type { PopoverProps } from '@hypothesis/frontend-shared/lib/components/feedback/Popover';
+import classnames from 'classnames';
+
+export type UserItem = {
+  username: string;
+  displayName: string | null;
+};
+
+export type MentionSuggestionsPopoverProps = Pick<
+  PopoverProps,
+  'open' | 'onClose' | 'anchorElementRef'
+> & {
+  /** List of suggestions to display */
+  suggestions: UserItem[];
+  /** Index for currently highlighted suggestion */
+  highlightedSuggestion: number;
+  /** Invoked when a suggestion is selected */
+  onSelectSuggestion: (selectedSuggestion: UserItem) => void;
+};
+
+/**
+ * A Popover component that displays a list of user suggestions.
+ */
+export default function MentionSuggestionsPopover({
```

Comment:
I think we could shorten the name here to `MentionPopover` and it would still be clear what it was for.